src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ExportRuleAttribute.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Rules
+{
+    /// <summary>
+    ///     Exports a XAML-based embedded rule.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Field)]
+    internal class ExportRuleAttribute : ExportPropertyXamlRuleDefinitionAttribute
```

Comment:
It used to be the case that sealed attributes were considerably faster to reflect upon ([CA1813](https://docs.microsoft.com/en-gb/visualstudio/code-quality/ca1813)). Not sure if that's still the case.

Reply:
In this case, its only affecting the tests. No code other than tests are reading these attributes via Reflection. MEF uses a metadata reader.